modules/xfem/test/tests/bimaterials/glued_ad_bimaterials_2d.i

```diff
@@ -0,0 +1,315 @@
+# This test is for two layer materials with different youngs modulus using AD
+# The global stress is determined by switching the stress based on level set values
+# The material interface is marked by a level set function
+# The two layer materials are glued together
+
+[GlobalParams]
+  order = FIRST
+  family = LAGRANGE
+  displacements = 'disp_x disp_y'
+[]
+
+[XFEM]
+  qrule = volfrac
+  output_cut_plane = true
+[]
+
+[UserObjects]
+  [./level_set_cut_uo]
+    type = LevelSetCutUserObject
+    level_set_var = ls
+  [../]
+[]
+
+[Mesh]
+  displacements = 'disp_x disp_y'
+  [gen]
+    type = GeneratedMeshGenerator
+    dim = 2
+    nx = 5
+    ny = 5
+    xmin = 0.0
+    xmax = 5.
+    ymin = 0.0
+    ymax = 5.
+    elem_type = QUAD4
+  []
+  [./left_bottom]
+    type = ExtraNodesetGenerator
+    new_boundary = 'left_bottom'
+    coord = '0.0 0.0'
+    input = gen
+  [../]
+  [./left_top]
+    type = ExtraNodesetGenerator
+    new_boundary = 'left_top'
+    coord = '0.0 5.'
+    input = left_bottom
+  [../]
+[]
+
+[AuxVariables]
+  [./ls]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+[]
+
+[AuxKernels]
+  [./ls_function]
+    type = FunctionAux
+    variable = ls
+    function = ls_func
+  [../]
+[]
+
+[Variables]
+  [./disp_x]
+  [../]
+  [./disp_y]
+  [../]
+[]
+
+[Functions]
+  [./ls_func]
+    type = ParsedFunction
+    value = 'y-2.5'
+  [../]
+[]
+
+[AuxVariables]
+  [./stress_xx]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./stress_yy]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./stress_xy]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./a_strain_xx]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./a_strain_yy]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./a_strain_xy]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./b_strain_xx]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./b_strain_yy]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+  [./b_strain_xy]
+    order = CONSTANT
+    family = MONOMIAL
+  [../]
+[]
+
+[Kernels]
```

Comment:
We should definitely be trying to switch everything over to the TM Master action.

Reply:
I added the use of the modules/ tensor mechanics action and retained the use of the individual stress and strain calculators classes for the A and B sides of the XFEM cut